crates/core/src/workflows/mod.rs

```diff
@@ -89,21 +78,160 @@ async fn try_make_local_validation_package(
     }
 }
 
-/// Gets hold of the validation package for the given entry.
-/// First tries to create it locally and if that fails will try to get the
-/// validation package from the source.
+/// Gets hold of the validation package for the given entry by trying several different methods.
 async fn validation_package(
     entry_with_header: &EntryWithHeader,
     context: Arc<Context>,
 ) -> Result<Option<ValidationPackage>, HolochainError> {
-    // 1. Try to construct it locally:
-    if let Ok(package) =
-        try_make_local_validation_package(&entry_with_header, context.clone()).await
+    // 0. Call into the DNA to get the validation package definition for this entry
+    // e.g. what data is needed to validate it (chain, entry, headers, etc)
+    let entry = &entry_with_header.entry;
+    let validation_package_definition = get_validation_package_definition(entry, context.clone())
```

Comment:
Not related to this PR, but why on earth does `get_validation_package_definition` return a `CallbackResult`?